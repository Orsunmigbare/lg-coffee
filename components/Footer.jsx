import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import SlimWrapper from './SlimWrapper';

const date = new Date();
const year = date.getFullYear();

const Footer = ({ links }) => (
  <footer>
    <Container fluid className="my-5">
      <SlimWrapper>
        {links.map((i) => (
          <Link href={i.path} key={i.title}>
            <h5 className="text-light m-2" style={{ cursor: 'pointer' }}>
              {i.title}
            </h5>
          </Link>
        ))}
      </SlimWrapper>
      <Row>
        <Col className="d-flex justify-content-center">
          <span className="text-light m-2" style={{ fontSize: '12px' }}>
            {year}
          </span>
        </Col>
      </Row>
    </Container>
  </footer>
);

Footer.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({}).isRequired).isRequired,
};

export default Footer;
