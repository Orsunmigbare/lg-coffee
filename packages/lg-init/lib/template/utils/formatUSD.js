/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  const string = JSON.stringify(number);
  const { length } = string;
  return JSON.parse(
    `${string.substring(0, length - 2)}.${string.substring(length - 2)}`
  ).toFixed(2);
};

export { formatUSD };
