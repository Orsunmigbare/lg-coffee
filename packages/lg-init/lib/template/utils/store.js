module.exports = {
  URL: process.env.LOCAL_URL || process.env.PUBLIC_URL,
  storeName: 'VisionSpeaks',
  logo: '/logo.png',
  accountOwner: 'Yulissa V',
  products: [
    {
      name: 'Lady In The Void',
      description: 'Holographic Sticker, This thing is bad ass',
      price: '287',
      sku: 'lad_vs_',
      images: [
        '/lady_in_the_void_TRANSP.png',
        '/lady_001.png',
        '/lady_002.png',
      ],
    },
    {
      name: 'Bunns In The Void',
      description: 'Holographic Sticker, This thing is bad ass',
      price: '287',
      sku: 'bun_vs_',
      images: [
        '/bunns_in_the_void_TRANSP.png',
        '/bunns_001.png',
        '/bunns_002.png',
      ],
    },
  ],
  pages: {
    globals: {
      // navigations
      navigation: [
        {
          title: 'Products',
          path: '/',
        },
        {
          title: 'Made To Order',
          path: '/info/contact',
        },
        {
          title: 'About',
          path: '/info/about',
        },
        {
          title: 'Cart',
          path: '/view-cart',
        },
      ],
      footer: [
        {
          title: 'Contact',
          path: 'mailto:support@visionspeaks.art',
        },
        {
          title: 'FAQ',
          path: '/info/faq',
        },
      ],
    },
    homePage: {
      hero: {
        image: '/heatpress.jpeg',
        headline: 'Made To Order',
        subHeadline: 'The highest quality work comes from custom work.',
        buttonName: 'Start Your Order',
        buttonPath: '/products',
        styles: {},
      },
    },
    info: {
      faq: [
        {
          q: 'Do you make custom products?',
          a: 'Yes',
        },
        {
          q: 'How long will it take to recieve my purchase?',
          a: 'Unless otherwise stated, All orders go out via Priority Mail.',
        },
        {
          q: 'What is your policy on refunds?',
          a: 'At this time, we are unable to honor refunds.',
        },
        {
          q:
            "I never received my purchase or something isn't right with my order. What can I do?",
          a: 'Reach out to support@visionspeaks.art.',
        },
      ],
      about: `
        VisionSpeaks is a custom art collective.
      `,
    },
  },
};
