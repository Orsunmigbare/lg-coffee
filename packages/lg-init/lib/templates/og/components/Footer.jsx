/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import SlimWrapper from './SlimWrapper';
import Contact from './ContactComponent';
import FAQComonent from './FAQComponent';
import AboutComponent from './AboutComponent';
import TermsAndConditions from './TermsAndConditions.mdx';
import ShippingAndReturns from './ShippingAndReturns.mdx';
import PrivacyPolicy from './PrivacyPolicy.mdx';
import { P, H2, PrimaryModal, Wrap } from './_helpers';

const date = new Date();
const year = date.getFullYear();

const Footer = ({ data }) => {
  const { theme, store } = data;
  return (
    <footer>
      <Container className="my-5">
        <SlimWrapper>
          <H2 theme={theme}>More Info</H2>
          <Wrap>
            <div className="d-flex flex-column">
              <PrimaryModal title="About" theme={theme}>
                <AboutComponent theme={theme} store={store} />
              </PrimaryModal>
              <PrimaryModal title="FAQ" theme={theme}>
                <FAQComonent store={store} theme={theme} />
              </PrimaryModal>
              <PrimaryModal title="Contact" theme={theme}>
                <Contact store={store} theme={theme} />
              </PrimaryModal>
            </div>
            <div className="d-flex flex-column">
              <PrimaryModal title="Terms and Conditions" theme={theme}>
                <TermsAndConditions />
              </PrimaryModal>
              <PrimaryModal title="Privacy Policy" theme={theme}>
                <PrivacyPolicy />
              </PrimaryModal>
              <PrimaryModal title="Shipping and Returns" theme={theme}>
                <ShippingAndReturns />
              </PrimaryModal>
            </div>
          </Wrap>
        </SlimWrapper>

        <Row>
          <Col className="d-flex justify-content-center">
            <P className="m-5 text-center" theme={theme}>
              Copyright <br /> {year}
            </P>
          </Col>
        </Row>
      </Container>
    </footer>
  );
};

Footer.propTypes = {
  data: PropTypes.shape({
    store: PropTypes.shape({}).isRequired,
    theme: PropTypes.shape({
      text: PropTypes.shape({}).isRequired,
      secondaryText: PropTypes.shape({}).isRequired,
      primaryBackground: PropTypes.shape({}).isRequired,
      secondaryBackground: PropTypes.shape({}).isRequired,
    }).isRequired,
  }).isRequired,
};

export default Footer;
