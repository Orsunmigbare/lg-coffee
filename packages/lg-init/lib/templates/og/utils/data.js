exports.data = {
  store: {
    name: 'VisionSpeaks',
    logo: '/logo.png',
    url: '',
    description: 'Custom sticker shop',
    contact: 'youremail@here.com',
    info: {
      faq: [
        {
          q: 'Do you make custom products?',
          a: 'Yes',
        },
        {
          q: 'How long will it take to recieve my purchase?',
          a: 'Unless otherwise stated, All orders go out via Priority Mail.',
        },
        {
          q: 'What is your policy on refunds?',
          a: 'At this time, we are unable to honor refunds.',
        },
        {
          q:
            "I never received my purchase or something isn't right with my order. What can I do?",
          a: 'Reach out to support@visionspeaks.art.',
        },
      ],
    },
    social: {
      facebook: '',
      twitter: '',
      instagram: '',
    },
  },
  theme: {
    text: { color: '#212529' },
    secondaryText: { color: '#6C757D' },
    primaryBackground: { backgroundColor: '#F8F9FA' },
    secondaryBackground: { backgroundColor: '#FFFFFF' },
  },
  products: {
    heading: 'New Arrivials',
    products: [
      {
        name: 'Bunns In The Void',
        description: 'Holographic Sticker, This thing is bad ass',
        price: 287,
        sku: 'bun_vs_',
        images: [
          '/bunns_in_the_void_TRANSP.png',
          '/bunns_001.png',
          '/bunns_002.png',
        ],
        type: 'sticker',
        qty: 40,
      },
      {
        name: 'Lady In The Void',
        description: 'Holographic Sticker, This thing is bad ass',
        price: 287,
        sku: 'lad_vs_',
        images: [
          '/lady_in_the_void_TRANSP.png',
          '/lady_001.png',
          '/lady_002.png',
        ],
        type: 'sticker',
        qty: 40,
      },
    ],
  },
};
