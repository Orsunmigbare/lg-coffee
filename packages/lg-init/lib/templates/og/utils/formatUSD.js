/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {     
  const { length } = number;
  const str = `${number.substring(0, length - 2)}.${number.substring(length - 2)}`;  
  return parseFloat(str).toFixed(2)
};

export { formatUSD };
