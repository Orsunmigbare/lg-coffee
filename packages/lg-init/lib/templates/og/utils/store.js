exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.png',
  hero: '/hero.png',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
