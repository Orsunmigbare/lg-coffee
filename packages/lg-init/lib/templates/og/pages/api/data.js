import { data } from '../../utils/data';

export default function handler(req, res) {
  res.statusCode = 200;
  res.setHeader('Content-Type', 'application/json');
  return res.end(JSON.stringify(data));
}
