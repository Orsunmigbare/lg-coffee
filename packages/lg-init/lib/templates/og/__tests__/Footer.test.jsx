import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Footer from '../components/Footer';
import { data } from '../utils/data';

it('Footer component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer data={data}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Footer component renders unchanged', () => {
  const tree = renderer.create(<Footer data={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});
