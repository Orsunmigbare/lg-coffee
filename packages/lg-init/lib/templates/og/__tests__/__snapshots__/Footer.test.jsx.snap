// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders unchanged 1`] = `
<footer>
  <div
    className="my-5 container"
  >
    <div
      className="align-items-center justify-content-center row"
    >
      <div
        className="col-lg-8 col-md-9 col-sm-12"
      >
        <h2
          className=""
          style={
            Object {
              "color": "#212529",
            }
          }
        >
          More Info
        </h2>
        <div
          className="d-flex flex-wrap"
        >
          <button
            className="btn btn-outline-dark mr-2 mt-2 lg_pill"
            id="About"
            onClick={[Function]}
            type="button"
          >
            About
          </button>
          <button
            className="btn btn-outline-dark mr-2 mt-2 lg_pill"
            id="FAQ"
            onClick={[Function]}
            type="button"
          >
            FAQ
          </button>
          <button
            className="btn btn-outline-dark mr-2 mt-2 lg_pill"
            id="Contact"
            onClick={[Function]}
            type="button"
          >
            Contact
          </button>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        <h6
          className="m-5"
          style={
            Object {
              "color": "#212529",
            }
          }
        >
          Copyright 
          2020
        </h6>
      </div>
    </div>
  </div>
</footer>
`;
